lib/kubernetes-deploy/kubectl.rb

```diff
@@ -24,40 +27,37 @@ def run(*args, log_failure: nil, use_context: true, use_namespace: true, output:
       raise_if_not_found: false, attempts: 1, output_is_sensitive: nil)
       log_failure = @log_failure_by_default if log_failure.nil?
       output_is_sensitive = @output_is_sensitive_default if output_is_sensitive.nil?
+      cmd = build_command_from_options(args, use_namespace, use_context, output)
+      current_attempt = 1
 
-      args = args.unshift("kubectl")
-      args.push("--namespace=#{@namespace}") if use_namespace
-      args.push("--context=#{@context}")     if use_context
-      args.push("--output=#{output}") if output
-      args.push("--request-timeout=#{@default_timeout}") if @default_timeout
-      out, err, st = nil
-
-      (1..attempts).to_a.each do |attempt|
-        @logger.debug("Running command (attempt #{attempt}): #{args.join(' ')}")
-        out, err, st = Open3.capture3(*args)
+      while current_attempt <= attempts
+        @logger.debug("Running command (attempt #{current_attempt}): #{cmd.join(' ')}")
+        out, err, st = Open3.capture3(*cmd)
         @logger.debug("Kubectl out: " + out.gsub(/\s+/, ' ')) unless output_is_sensitive
 
         break if st.success?
+        raise(ResourceNotFoundError, err) if err.match(NOT_FOUND_ERROR_TEXT) && raise_if_not_found
 
         if log_failure
-          @logger.warn("The following command failed (attempt #{attempt}/#{attempts}): #{Shellwords.join(args)}")
+          escaped_cmd = Shellwords.join(cmd)
+          @logger.warn("The following command failed (attempt #{current_attempt}/#{attempts}): #{escaped_cmd}")
           @logger.warn(err) unless output_is_sensitive
-        end
-
-        if err.match(NOT_FOUND_ERROR_TEXT)
-          raise(ResourceNotFoundError, err) if raise_if_not_found
         else
           @logger.debug("Kubectl err: #{err}") unless output_is_sensitive
```

Comment:
should we log something in debug mode?

Reply:
What do you mean? That's what this line does

Reply:
I think I wanted to say: Should we log something in debug mode _if the output_is_sensitive_ ? 
`@logger.debug("Kubectl err: #{output_is_sensitive  ? '<sensitive>': err}")`

Reply:
Ah! sure, we can do that